public/javascripts/stopsTable.js

```diff
@@ -0,0 +1,164 @@
+		// Get the button stop name, also case insensitive.
+		buttonElem = $(buttonElem);
+		let stopName = buttonElem.text().toLowerCase();
+
+		// If the button's stop name includes the search term, show this
+		// button in case it was hidden before. Otherwise, hide this button.
+		if (stopName.includes(searchTerm)) {
+			buttonElem.show();
+		} else {
+			buttonElem.hide();
+		}
+	}
+}
+
+/**
+ * Updates the list of stops displayed in the sidebar in accord with the
+ * stopTransports XML file fetched from TriMet's servers and goes on to hide
+ * all available stops until a search is entered.
+ */
+function updateStops() {
+	let xml = staticFetch.data.stopTransports;
+
+	// Get and clear out our current list of routes.
+	let stopsElem = $("#stops-list").empty();
+
+	// Loop over all the stops of the root of the XML document and add
+	// buttons for each of them.
+	for (let stopNode of xml.documentElement.getElementsByTagName("Placemark")) {
+		let buttonElem = $("<button>")
+			.text(stopNode.getElementsByTagName("name")[0].textContent)
+			.on("click", e => {
+				// When the button is clicked, show the routes that go through that stop.
+				stopPages.showTab("transport");
+				updateTransport(stopNode);
+			});
+
+		stopsElem.append(buttonElem);
+	}
+
+	//After creating all button elements, hides the list in order to prevent wait time issues.
+	stopsElem.hide();
+
+	// Since the list of buttons was changed, also re-update which ones are
+	// shown due to the search bar.
+	updateStopSearch();
+}
+
+// When the search bar is edited and enter is clicked, updates stop list with relevant
+// stops based on search entered.
+$("#stops-search").keyup(function(e) {
+	if (e.keyCode == 13) {
+		$("#stops-list").show();
+		updateStopSearch();
+	}
+});
+
+// Clear and refocus the search field when clicking the X button, also 
+// hides all stop buttons again for wait times. 
+$("#stops-clear-search").on("click", e => {
+	$("#stops-search").val("").focus();
+	//Hides the list of stops again if there is no search input to prevent 
+	//incredibly long list / wait times.
+	$("#stops-list").hide();
+	updateStopSearch();
+});
+
+// Goes back to the list of stops found from the given search
+$("#transport-back").on("click", e => {
+	stopPages.showTab("stops");
+});
+
+/* Fetches every possible route and the route's stops from TriMet and 
+ * stores it within the "transportRoutes" variable in order to prevent 
+ * fetching from happening everytime a search is made. 
+ */
+staticFetch.addData("transportRoutes",
+	() => fetchAppXml("https://developer.trimet.org/ws/V1/routeConfig", {
+		stops: true,
+		dir: true,
+	})
+);
+
+/* Fetches every possible stop from TriMet and stores it within the "stopTransports" variable 
+ * in order to prevent fetching from happening everytime a search is made. 
+ */
+staticFetch.addData("stopTransports",
+	() => fetchAppXml("https://developer.trimet.org/gis/data/tm_stops.kml", {
+		ll: true,
+	})
+);
```

Comment:
`stops.js` already fetches these URLs as "routeStops" and "stopCoords".  If it's fetched somewhere else, you don't need to do it again.